### PR TITLE
REP-5329 Teach retryer to run +1 callback in parallel.

### PR DESCRIPTION
**Pull request by FGasper**
We need retry logic around `find` cursor iteration, but the retryer’s IterationSuccess() method won’t work properly with how we compare partitions document-by-document. If, for example, the source hangs but we call IterationSuccess() on every destination success, we’ll never time out the retries. But we need IterationSuccess() to work or else there are no retries for partitions that take a long time to read.

This changeset fixes that by rewriting the retryer to take multiple callbacks. Each callback now takes a context that the retryer uses to stop the function whenever another thread has failed.

Since migration-verifier doesn’t handle DDL events, there’s no point to handling collection UUIDs. Thus, the initial commits in this PR resolve longstanding technical debt.

This also synchronizes migration-verifier’s internal list of transient errors with mongosync’s.

### CODE REVIEW

File: internal/verifier/compare.go [hunk 1]
```diff
@@ -308,16 +315,26 @@ func (verifier *Verifier) getFetcherChannels(
...
+	cursor *mongo.Cursor,
+	writer chan<- bson.Raw,
+) error {
 	for cursor.Next(ctx) {
+		state.NoteSuccess()
 		writer <- slices.Clone(cursor.Current)
 	}
 
+	if cursor.Err() == nil {
+		state.NoteSuccess()
```
Comment (tdq45gj) [inline internal/verifier/compare.go:335]: Looks like the retryer refreshes `lastResetTime` once the error group finishes. Is it redundant to call `NoteSuccess` before the callback function returns?

**Review by tdq45gj — APPROVED** [review]
LGTM % one small question

File: internal/verifier/compare.go (hunk 1 above)
Comment (FGasper) [reply internal/verifier/compare.go:335]: Probably. Removed.

**Review by edobranov — APPROVED** [review]
LGTM

**Pull request merged**